torch/nn/parallel/distributed.py

```diff
@@ -748,6 +751,14 @@ def forward(self, *inputs, **kwargs):
                     work,
                     self.ddp_uneven_inputs_config.ddp_join_divide_by_initial_world_size,
                 )
+                if self.ddp_uneven_inputs_config.ddp_join_throw_on_early_termination:
+                    zeros = torch.zeros(1, device=self.device)
+                    dist.all_reduce(zeros, group=self.process_group)
+                    should_throw_stop_iteration = zeros.item()
```

Comment:
Are you expecting the `all_reduce` result to become 1 in any case? Shouldn't it be always 0s by adding up multiple 0s?